internal/modules/circle/circle_report/service.go

```diff
@@ -0,0 +1,30 @@
+package circle_report
+
+import (
+	"catalog-be/internal/domain"
+	"catalog-be/internal/entity"
+)
+
+type CircleReportService struct {
+	repo *CircleReportRepo
+}
+
+// Initialize Circle Report Service
+func NewCircleReportService(repo *CircleReportRepo) *CircleReportService {
+	return &CircleReportService{repo}
+}
+
+// CreateReport implements CircleReportService
+func (c *CircleReportService) CreateCircleReport(circleID int, userID int, reason string) *domain.Error {
+	return c.repo.CreateReport(circleID, userID, reason)
+}
```

Comment:
```suggestion
func (c *CircleReportService) CreateCircleReport(report *entity.Report) *domain.Error {
	return c.repo.CreateReport(circleID, userID, reason)
}
```
sepertinya bisa langsung oper entity ya biar rapih dan ga panjang parameternya